src/Scientist/FireAndForgetResultPublisher.cs

```diff
@@ -0,0 +1,58 @@
+﻿using GitHub.Internals;
+using System;
+using System.Threading.Tasks;
+
+namespace GitHub
+{
+    public enum ResultPublisherExceptionBehavior
+    {
+        Rethrow,
+        Ignore
+    }
+
+    public class FireAndForgetResultPublisher : IResultPublisher
+    {
+        private readonly IResultPublisher _publisher;
+        private readonly Action<Exception> _onPublisherException;
+        private readonly ConcurrentSet<Task> _publishingTasks = new ConcurrentSet<Task>();
+        private readonly ResultPublisherExceptionBehavior _exceptionBehavior;
+
+        public FireAndForgetResultPublisher(IResultPublisher publisher, ResultPublisherExceptionBehavior exceptionBehavior = ResultPublisherExceptionBehavior.Ignore, Action<Exception> onPublisherException = null)
```

Comment:
Optional arguments in public APIs can cause versioning issues down the road: http://haacked.com/archive/2010/08/10/versioning-issues-with-optional-arguments.aspx/

We should use constructor overloads instead.